src/wp-includes/blocks.php

```diff
@@ -903,10 +904,28 @@ function make_after_block_visitor( $hooked_blocks, $context ) {
 				? $hooked_blocks[ $anchor_block_type ][ $relative_position ]
 				: array();
 
+		if ( isset( $block['attrs']['layout'] ) ) {
+			$layout = $block['attrs']['layout'];
+		}
+
 		/** This filter is documented in wp-includes/blocks.php */
 		$hooked_block_types = apply_filters( 'hooked_block_types', $hooked_block_types, $relative_position, $anchor_block_type, $context );
 		foreach ( $hooked_block_types as $hooked_block_type ) {
-			$markup .= get_hooked_block_markup( $block, $hooked_block_type );
+			$attributes = array();
+
+			// Does the anchor block have a layout attribute?
+			if ( isset( $layout ) ) {
+				// Has the hooked block type opted into layout block support?
+				$hooked_block_type_obj = WP_Block_Type_Registry::get_instance()->get_registered( $hooked_block_type );
+				if ( $hooked_block_type_obj && $hooked_block_type_obj instanceof WP_Block_Type ) {
+					if ( $hooked_block_type_obj->attributes && isset( $hooked_block_type_obj->attributes['layout'] ) ) {
```

Comment:
Not all blocks that support `layout` have the attribute explicitly declared; it only appears when the block has a non-default layout configuration. A more reliable check would be `block_has_support( $block_type, 'layout', false )`.